x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/fields/ExtractedField.java

```diff
@@ -112,6 +112,7 @@ public static ExtractedField newField(String alias, String name, ExtractionMetho
             }
             if (value[0] instanceof String) { // doc_value field with the epoch_millis format
                 value[0] = Long.parseLong((String) value[0]);
+                //TODO confirm if can remove the script time field
```

Comment:
OK, I had a discussion with @jpountz and got to the bottom of this. `BaseDateTime` was the type returned for date fields in doc values prior to allowing configuring the doc value format. Thus, we still need this for mixed 6.x clusters. The comment below `// script field` is actually misleading. ML never supported scripted date fields.

Reply:
OK, I think the only thing left is to remove this TODO and correct the comment below (`// script field`). A more accurate comment would be ` // for handling doc value type prior to 6.4`

Reply:
I need to investigate what will be the behaviour for a mix cluster like that. 
Is there a test in any of the old branches that I can modify and see that place being hit? We could possibly then think of creating a similar one on the master branch with a mixed cluster?

Reply:
Actually we should be able to remove this from `master`. `7.x` nodes should only talk to `6.latest` which already have the doc value format specified. It's only in `6.x` we need to keep this code for. Any datafeed test should be hitting this prior to `6.4`.

Reply:
Note that prior to `6.4` there was no `if` block in this method. It was directly casting to the joda time.

Reply:
See https://github.com/elastic/elasticsearch/pull/31463

Reply:
as we discussed, that sounds really good. Will remove that line in master